### PR TITLE
Fix build on macOS

### PR DESCRIPTION
**Pull request by moonchen**
- Use -undefined dynamic_lookup to ignore unresolved dependencies which are located inside traffic_server
- Add some missing dependencies
- Break a cyclic dependency between tscppapi and tscore

### CODE REVIEW

File: src/tscpp/api/CMakeLists.txt [hunk 1]
```diff
@@ -45,5 +45,10 @@ add_library(ts::tscppapi ALIAS tscppapi)
 target_link_libraries(tscppapi
     PUBLIC
     libswoc
-    yaml-cpp::yaml-cpp)
+    yaml-cpp::yaml-cpp
+    )
```
Comment (JosiahWI) [inline src/tscpp/api/CMakeLists.txt]: Dedent.

File: src/tscpp/api/CMakeLists.txt [hunk 2]
```diff
@@ -45,5 +45,10 @@ add_library(ts::tscppapi ALIAS tscppapi)
...
 install(TARGETS tscppapi)
+
+if (APPLE)
+    set_target_properties(tscppapi PROPERTIES LINK_FLAGS "-undefined dynamic_lookup")
```
Comment (JosiahWI) [inline src/tscpp/api/CMakeLists.txt]:
```suggestion
    # does this need to be PUBLIC I wonder?
    target_link_options(tscppapi PUBLIC "-undefined dynamic_lookup")
```

File: src/tscpp/util/CMakeLists.txt [hunk 3]
```diff
@@ -25,7 +25,8 @@ add_library(ts::tscpputil ALIAS tscpputil)
 target_link_libraries(tscpputil
 	PUBLIC
 		libswoc
-		yaml-cpp::yaml-cpp)
+		yaml-cpp::yaml-cpp
+                ts::tscore)
```
Comment (JosiahWI) [inline src/tscpp/util/CMakeLists.txt]: Closing paren on newline and dedent.

**Review by JosiahWI — CHANGES_REQUESTED** [review]
Thank you so much for figuring out how to fix this. It will be awesome to have the Mac build working again.

File: src/tscpp/api/CMakeLists.txt (hunk 1 above)
Comment (moonchen) [reply src/tscpp/api/CMakeLists.txt]: Fixed

File: src/tscpp/api/CMakeLists.txt (hunk 2 above)
Comment (moonchen) [reply src/tscpp/api/CMakeLists.txt]: Fixed

File: src/tscpp/util/CMakeLists.txt (hunk 3 above)
Comment (moonchen) [reply src/tscpp/util/CMakeLists.txt]: Fixed

**Review by JosiahWI — APPROVED** [review]

File: src/tscpp/api/CMakeLists.txt (hunk 2 above)
Comment (moonchen) [reply src/tscpp/api/CMakeLists.txt]: It could be PUBLIC, but the plugins still need to be dynamic_lookup because they're plugins.

**Pull request merged**